### ISSUE TITLE
[question] find_min_global repeats argument choices if set to use integer variables

### ISSUE THREAD

**Issue by lesshaste** [davisking/dlib#2542]
dlib version 19.23.0 on ubuntu 20.04 64 bit using dlib from python

When using find_min_global with is_integer_variable  set to True find_min_global repeats the same parameters at times. Is this expected?  It is not a big problem as I can just cache the results of the objective function to make it fast if you call it with the same arguments twice or more but it doesn't seem ideal.

**Comment by davisking** [comment]
Yeah that can happen.  If it matters you should do some caching inside the
function being optimized.
